### PR TITLE
Rebase attempt for v3.64.2

### PR DESCRIPTION
**Pull request by DavidMah**
I got this by checking out 3.64.2 and then successively applying cherry picks and trying to migrate the code.

This is the list I cherry picked on
```
  * d517c589a4    David Mah   Wed Jul 28 10:48:32 2021 -0700  ListenerRule: Do not replace for priority tweaks (#296)
  * 35bb30f0fa    David Mah   Tue Jul 27 17:00:48 2021 -0700  ForceNew for task set network configuration
  * ef37806ab6    John Firebaugh  Mon Jul 19 16:40:13 2021 -0700  Automate releases to figma/aws on terraform registry
  * f10752ef01    John Firebaugh  Wed Sep 23 11:18:00 2020 -0700  Don't set network_configuration property on services with EXTERNAL deployment controller
  * 9dc5c3acb6    Kumarappan-A    Mon May 20 23:36:26 2019 -0700  resource/aws_ecs_task_set: Support ECS task set
```

Then I updated code until `go build` worked

```
got it to [~/figma/figma-terraform-provider-aws] ➜ go build
[~/figma/figma-terraform-provider-aws] ➜
```

# TODO

Still building this to v.3.64.2-figma.2, need to try it on main/ in a dev cluster

### CODE REVIEW

File: internal/service/ecs/service.go [hunk 1]
```diff
@@ -721,8 +721,10 @@ func resourceServiceRead(d *schema.ResourceData, meta interface{}) error {
 		log.Printf("[ERR] Error setting placement_constraints for (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("network_configuration", flattenEcsNetworkConfiguration(service.NetworkConfiguration)); err != nil {
-		return fmt.Errorf("error setting network_configuration for (%s): %w", d.Id(), err)
+	if service.DeploymentController == nil || aws.StringValue(service.DeploymentController.Type) != ecs.DeploymentControllerTypeExternal {
```
Comment (DavidMah) [inline internal/service/ecs/service.go:724]: this is copy pasted in from current code at https://github.com/figma/terraform-provider-aws/blob/figma/aws/resource_aws_ecs_service.go#L722

File: internal/service/ecs/task_set.go [hunk 2]
```diff
@@ -0,0 +1,744 @@
+package ecs
```
Comment (DavidMah) [inline internal/service/ecs/task_set.go:1]: The diff between this and the current file is at https://gist.github.com/DavidMah/c7992d8fc651d1e323536300bd3a2b4a

Library functions are renamed, but no logic has been rearranged other than changing the callsites to those library functions

File: internal/service/elbv2/listener_rule.go [hunk 3]
```diff
@@ -55,10 +55,11 @@ func ResourceListenerRule() *schema.Resource {
 				ValidateFunc: verify.ValidARN,
 			},
 			"priority": {
-				Type:         schema.TypeInt,
-				Optional:     true,
-				Computed:     true,
-				ForceNew:     true,
+				Type:     schema.TypeInt,
```
Comment (DavidMah) [inline internal/service/elbv2/listener_rule.go:58]: Copy pasted from current code at https://github.com/figma/terraform-provider-aws/blob/figma/aws/resource_aws_lb_listener_rule.go#L46

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,54 +1,51 @@
-name: Post Publish
+# This GitHub action can publish assets for release when a tag is created.
```
Comment (DavidMah) [inline .github/workflows/release.yml:1]: this is copy pasted from the current code, there is no delta

**Review by jfirebaugh — APPROVED** [review]
LGTM. I take it you abandoned your other ALB changes (https://github.com/hashicorp/terraform-provider-aws/commit/1e5aa40a3e94c09a885454b51b535cfe448953ea and so on)?

Before pushing, please massage the history into a series of self-contained commits, to make the next rebase as easy as possible.

**Comment by DavidMah** [comment]
> LGTM. I take it you abandoned your other ALB changes ([hashicorp@1e5aa40](https://github.com/hashicorp/terraform-provider-aws/commit/1e5aa40a3e94c09a885454b51b535cfe448953ea) and so on)?
> 
> Before pushing, please massage the history into a series of self-contained commits, to make the next rebase as easy as possible.

yep thats abandoned, was encountering some quirks in testing it and figured to not pursue it for now.

massaging history sounds good!

**Comment by DavidMah** [comment]
moved changes into `figma` branch, now working from there

**Pull request closed without merging**